nntrainer/src/layers.cpp

```diff
@@ -315,17 +241,25 @@ int OutputLayer::setCost(CostType c) {
   return status;
 }
 
-Tensor OutputLayer::forwarding(Tensor in) {
+Tensor FullyConnectedLayer::forwarding(Tensor in, int &status) {
   input = in;
   hidden = input.dot(weight).add(bias);
+
+  if (this->bn_fallow)
+    return hidden;
+
   if (activation_type == ACT_SOFTMAX) {
     return hidden.apply(softmax);
   } else {
     return hidden.apply(activation);
   }
 }
 
-Tensor OutputLayer::forwarding(Tensor in, Tensor output) {
+Tensor FullyConnectedLayer::forwarding(Tensor in, Tensor output, int &status) {
+  if (!this->last_layer) {
+    ml_loge("Error: Cannot update cost. This is not last layer of network");
+    status = ML_ERROR_INVALID_PARAMETER;
```

Comment:
Should there be `return` because it is an error?

Reply:
Tensor should be returned to maintain whole process. We are going to use parameter return.